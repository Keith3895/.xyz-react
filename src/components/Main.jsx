import React, { Component } from 'react';
import { Switch, Route, Router } from 'react-router-dom';

class Main extends Component {
    render() {
        return (
            <div>
                <Route exact path='/' render={(props) => {
                    return (<h1>Home</h1>);
                }} />
                <Route path='/project' render={(props) => {
                    return (<h1>Project</h1>);
                }} />
            </div>
        );
    }
}
export default Main;