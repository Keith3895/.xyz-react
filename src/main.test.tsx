import { act, render, screen, waitFor } from '@testing-library/react';
import { createHashRouter, RouterProvider } from 'react-router-dom';
import Main from './main';
import renderer from 'react-test-renderer';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    RouterProvider: jest.fn(),
}));

jest.mock('./components/nav', () => () => <div>NavBar</div>);
jest.mock('./components/footer', () => () => <div>Footer</div>);

describe('Main', () => {
    it('renders NavBar, RouterProvider and Footer', () => {
        render(<Main />);
        expect(screen.getByText('NavBar')).toBeInTheDocument();
        expect(screen.getByText('Footer')).toBeInTheDocument();
        expect(RouterProvider).toHaveBeenCalled();
    });


    it('renders correctly when there are no items', () => {
        const tree = renderer.create(<Main />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});