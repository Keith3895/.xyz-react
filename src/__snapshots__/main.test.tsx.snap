// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly when there are no items 1`] = `
<div>
  <header>
    <div>
      NavBar
    </div>
  </header>
  <div
    className="mt-16 pt-2"
  />
  <div>
    Footer
  </div>
</div>
`;
